pkg/controller/vsphere/actuator.go

```diff
@@ -76,7 +79,14 @@ func (a *Actuator) Create(ctx context.Context, machine *machinev1.Machine) error
 		return a.handleMachineError(machine, fmtErr, createEventAction)
 	}
 	a.eventRecorder.Eventf(machine, corev1.EventTypeNormal, createEventAction, "Created Machine %v", machine.GetName())
-	return scope.PatchMachine()
+
+	if err := scope.PatchMachine(); err != nil {
+		return err
+	}
+	// Return a requeue after error to add a brief delay between reconciles
+	// otherwise we might get a stale object from cache without a taskID and
+	// issue a double create.
+	return &machinecontroller.RequeueAfterError{RequeueAfter: requeueAfterSeconds * time.Second}
```

Comment:
This won't result in the next reconcile taking 20 seconds, as the patch update from the API server will put another reconcile request in the queue, hopefully this happening is with the updated version.  I will probably transform this comment into a code comment before merging.

Reply:
i'm having a little trouble following this part, it seems like we always return an error. does this return value force a requeue or only if an error (another error?) arises?

Reply:
It's non-obvious without knowing what the other side of the Actuator interface (machine controller) is doing: https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/machine/controller.go#L315-L327

Before this patch, we'd return nil, which would instruct the machine-controller to return an error which resulted in an immediate requeue.

Here, we return a specific class of error, which results in the machine-controller requesting a delayed re-queue.  This will prevent the machine-controller's artificial error (meant to immediately requeue) from front-running the queue event generated by patching the object, at least that's the theory.

Reply:
yea, definitely confusing from looking at the code. i think i am starting to get it though, thanks for the explanation